### PR TITLE
[ci] Correct windows build and test in CI

### PR DESCRIPTION
**Pull request by Leonz5288**
Related issue = #

### CODE REVIEW

File: misc/ci_setup.py [hunk 1]
```diff
@@ -255,23 +255,25 @@ def run(self):
...
             execute_command('echo $PYTHONPATH')
-        elif get_os_name() != 'win':
+        else:
             # compile ..
             arg = environ.get('CI_SETUP_CMAKE_ARGS', '')
             os.makedirs('build', exist_ok=True)
-            execute_command(
-                f'TAICHI_CMAKE_ARGS="{arg}" {sys.executable} setup.py install --user'
-            )
+            if get_os_name() != 'win':
```
Comment (k-ye) [inline misc/ci_setup.py:262]: set `TAICHI_CMAKE_ARGS={args}` as a new env variable, then we can unify this part?

File: .github/workflows/presubmit.yml [hunk 2]
```diff
@@ -217,10 +217,10 @@ jobs:
           clang --version
           cd D:\a\taichi\taichi
           python -m pip install --user numpy pybind11 gitpython
-          $env:CI_SETUP_CMAKE_ARGS = "-G'Visual Studio 16 2019' -A x64 -DPYTHON_EXECUTABLE='$env:PYTHON' -DLLVM_DIR='C:\taichi_llvm\lib\cmake\llvm'"
           python misc/ci_setup.py ci
         env:
           PYTHON: C:\hostedtoolcache\windows\Python\3.7.9\x64\python.exe
+          CI_SETUP_CMAKE_ARGS: -A x64 -DLLVM_DIR='C:\taichi_llvm\lib\cmake\llvm'
```
Comment (k-ye) [inline .github/workflows/presubmit.yml:223]: Did you drop `-G'Visual Studio 16 2019'` intentionally?

Comment (Leonz5288) [reply .github/workflows/presubmit.yml:223]: Correct, this is done by default with the msbuild path added. Also, passing this with spaces leads to confusion when parsing cmake_args.

File: setup.py [hunk 3]
```diff
@@ -127,7 +127,8 @@ def run(self):
         cmake_args += ['-DCMAKE_BUILD_TYPE=' + cfg]
 
         # Assuming Makefiles
-        build_args += ['--', f'-j{multiprocessing.cpu_count()}']
+        if get_os_name() != 'win':
```
Comment (ailzhang) [inline setup.py:130]: sorry I'm not familiar with windows, why do we want to specialize this? ;)

Comment (Leonz5288) [reply setup.py:130]: This is simply because MSBuild does not support "-j{number}" switch for multiprocessor build.

**Review by ailzhang — APPROVED** [review]
Looks great! Thanks @Leonz5288 !

File: setup.py [hunk 4]
```diff
@@ -94,8 +94,9 @@ def run(self):
 class CMakeBuild(build_ext):
     def parse_cmake_args_from_env(self):
         # Source: TAICHI_CMAKE_ARGS=... python setup.py ...
+        import shlex
```
Comment (k-ye) [inline setup.py:97]: This is great :-)

**Review by k-ye — APPROVED** [review]

**Comment by k-ye** [comment]
Hmm, need to update Appveyor as well? (https://ci.appveyor.com/project/yuanming-hu/taichi/build/job/8j39ds90h03fli5r)

https://github.com/taichi-dev/taichi/blob/master/appveyor.yml

**Pull request merged**